packages/mdc-segmented-button/segmented-button/component.ts

```diff
@@ -69,6 +69,14 @@ export class MDCSegmentedButton extends MDCComponent<MDCSegmentedButtonFoundatio
       segment.setIndex(index);
       segment.setIsSingleSelect(isSingleSelect);
     });
+
+    const selectedSegments = this.segments_.filter((segment) => segment.isSelected());
```

Comment:
I think it is not required modify the DOM to force select single segmented on initialization.

It might be helpful if component throws error when a component has multi selected segments for single select variant.

WDYT?

Reply:
Throwing an error makes sense so that we don't surprise the user with potentially unintuitive behavior.